workflows/models.py

```diff
@@ -195,7 +195,7 @@ def get_or_create_analysis(self, update=False):
 
             if set(analysis['input_datasets']) != set(self.input_datasets):
                 if update:
-                    tantalus_api.update('analysis', id=analysis['id'], input_datasets=input_datasets)
+                    tantalus_api.update('analysis', id=analysis['id'], input_datasets=self.input_datasets)
```

Comment:
I think we need to grab input datasets from the analysis object in general so maybe dont store as self.input_datasets